src/datadog/trace_segment.cpp

```diff
@@ -202,50 +225,45 @@ void TraceSegment::update_decision_maker_trace_tag() {
 
   assert(sampling_decision_);
 
+  const auto found = std::find_if(
```

Comment:
If this code gets refactored in the future, some special care will be required for the points where `found` or `value` become invalid. A comment for future-self at those points might be a good idea.

Reply:
I'll add comments.